modules/services/window-managers/i3-sway/i3.nix

```diff
@@ -143,51 +143,46 @@ let
     floatingCriteriaStr windowCommandsStr colorSetStr windowBorderString
     fontConfigStr keybindingDefaultWorkspace keybindingsRest workspaceOutputStr;
 
-  startupEntryStr = { command, always, notification, workspace, ... }: ''
-    ${if always then "exec_always" else "exec"} ${
-      if (notification && workspace == null) then "" else "--no-startup-id"
-    } ${
-      if (workspace == null) then
+  startupEntryStr = { command, always, notification, workspace, ... }:
+    concatStringsSep " " [
+      (if always then "exec_always" else "exec")
+      (if (notification && workspace == null) then "" else " --no-startup-id")
+      (if (workspace == null) then
         command
       else
-        "i3-msg 'workspace ${workspace}; exec ${command}'"
-    }
-  '';
+        "i3-msg 'workspace ${workspace}; exec ${command}'")
+    ];
 
   configFile = pkgs.writeText "i3.conf" ((if cfg.config != null then
-    with cfg.config; ''
-      ${fontConfigStr fonts}
-      floating_modifier ${floating.modifier}
-      ${windowBorderString window floating}
-      hide_edge_borders ${window.hideEdgeBorders}
-      force_focus_wrapping ${if focus.forceWrapping then "yes" else "no"}
-      focus_follows_mouse ${if focus.followMouse then "yes" else "no"}
-      focus_on_window_activation ${focus.newWindow}
-      mouse_warping ${if focus.mouseWarping then "output" else "none"}
-      workspace_layout ${workspaceLayout}
-      workspace_auto_back_and_forth ${
+    with cfg.config;
+    concatStringsSep "\n" ([
+      (fontConfigStr fonts)
+      "floating_modifier ${floating.modifier}"
+      (windowBorderString window floating)
+      "hide_edge_borders ${window.hideEdgeBorders}"
+      "force_focus_wrapping ${if focus.forceWrapping then "yes" else "no"}"
```

Comment:
Maybe it's worth extracting this out to a function while you are at it. See the `yesno` function in the neomutt module.

We could probably extract it out to lib as well, but I'm fine with doing that in a separate PR.

Reply:
Created a follow up #2772 for it, as it would be done in a another PR anyway I guess...